src/utils/contracts.ts

```diff
@@ -57,6 +59,36 @@ export const getFlashMintLeveragedContract = (
   )
 }
 
+export const getIndexFlashMintLeveragedContractAddress = (
+  chainId = ChainId.Mainnet
+) => {
+  switch (chainId) {
+    default:
+      return FlashMintLeveragedAddress
+  }
+}
+
+/**
+ * Returns an instance of the Index FlashMintLeveraged contract (mainnet)
+ *
+ * @param signerOrProvider  a signer or provider
+ * @param chainId           chainId for contract (default mainnet)
+ *
+ * @returns an instance of a FlashMintLeveraged contract
+ */
+export const getIndexFlashMintLeveragedContract = (
+  signerOrProvider: Signer | Provider | undefined,
+  chainId: number = ChainId.Mainnet
+): Contract => {
+  const contractAddress = getIndexFlashMintLeveragedContractAddress(chainId)
+  return new Contract(
+    contractAddress,
+    // FIXME: add new ABI? might be the same?
+    EXCHANGE_ISSUANCE_LEVERAGED_ABI,
```

Comment:
This `FIXME` can probably be deleted? Or should there be a new ABI?

Reply:
there is technically a new abi due to the deployment options being different but this should work for encoding all function calls that we care about.